presentation/json.go

```diff
@@ -0,0 +1,140 @@
+		s.Incremental.PacketsPassedFilter, s.Incremental.ResponsesParsed, u.dropLabel(*s.Incremental),
+		s.Culumative.PacketsPassedFilter, s.Culumative.ResponsesParsed, u.dropLabel(*s.Culumative)))
+
+	if f != nil {
+		// Write to the output file if specified
+		f.WriteString(string(reportJson) + "\n")
+	} else {
+		u.Log(string(reportJson))
+	}
+
+	return nil
+}
+
+func (u *uiContext) openFile(filename string) (*os.File, error) {
+	if filename != "" {
+		f, err := os.OpenFile(u.outputFile, os.O_RDWR|os.O_CREATE|os.O_APPEND|os.O_SYNC, 0644)
+		return f, err
+	} else {
+		return nil, nil
+	}
+}
+
+func formatReportAsJson(report analysis.Report, stats StatsSet, totalKeys int, totalBandwidth int64, reportedKeys int, reportedBandwidth int64) ([]byte, error) {
+	reportMap := map[string]interface{}{}
+	reportMap["ts"] = report.Timestamp.UTC().Unix()
+	reportMap["ts_s"] = report.Timestamp.String()
+	reportMap["totalKeys"] = totalKeys
+	reportMap["totalBandwidth"] = totalBandwidth
+	reportMap["reportedKeys"] = reportedKeys
+	reportMap["reportedBandwidth"] = reportedBandwidth
+	reportMap["reportedBandwidthPercentage"] = 100*float64(reportedBandwidth)/float64(totalBandwidth)
+	reportMap["rows"] = reportToList(report)
+	reportMap["stats"] = statsSetToMap(stats)
+	return json.Marshal(reportMap)
+}
+
+func reportToList(report analysis.Report) []map[string]interface{} {
+	reportList := make([]map[string]interface{}, len(report.Rows))
+	for idx, row := range report.Rows {
+		reportList[idx] = reportRowToMap(row, report.KeyColNames, report.ValColNames)
+	}
+	return reportList
+}
+
+func reportRowToMap(reportRow analysis.ReportRow, keyColNames []string, valueColNames []string) map[string]interface{} {
+	rowMap := map[string]interface{}{}
+
+	for idx, keyCol := range keyColNames {
+		rowMap[keyCol] = reportRow.Key[idx]
+	}
+
+	for idx, valueCol := range valueColNames {
+		rowMap[valueCol] = reportRow.Values[idx]
+	}
+
+	return rowMap
+}
+
+func statsSetToMap(stats StatsSet) map[string]interface{} {
+	statsSetMap := map[string]interface{}{}
+	statsSetMap["incremental"] = statsToMap(*stats.Incremental)
+	return statsSetMap
+}
+
+func statsToMap(stats Stats) map[string]interface{} {
+	statsMap := map[string]interface{}{}
+	statsMap["PacketsEnteredFilter"] = stats.PacketsEnteredFilter
+	statsMap["PacketsPassedFilter"] = stats.PacketsPassedFilter
+	statsMap["PacketsCaptured"] = stats.PacketsCaptured
+	statsMap["PacketsDroppedKernel"] = stats.PacketsDroppedKernel
+	statsMap["PacketsDroppedParser"] = stats.PacketsDroppedParser
+	statsMap["PacketsDroppedAnalysis"] = stats.PacketsDroppedAnalysis
+	statsMap["PacketsDroppedTotal"] = stats.PacketsDroppedTotal
+	statsMap["ResponsesParsed"] = stats.ResponsesParsed
+	return statsMap
+}
+
+// Given a slice of rows, return the sum of the sum(size) columns
+func totalBytesUseForKeys(rows []analysis.ReportRow) int64 {
+	total := int64(0)
+	for _,r := range rows {
+		total = total + r.Values[1]
```

Comment:
This badness is especially bad. Can we at least use constants instead of magic numbers?

Reply:
Just following the existing badness :)